src/internal/lib/asset-base-url.ts

```diff
@@ -10,6 +10,28 @@ export default () => {
     return result;
   }
 
+  if (gateway.requiredSettings[0] === "X-GP-Api-Key") {
+    const env = gateway.getEnv(options);
+    switch (env) {
+      case "local":
+        return `http://localhost:8080/v${version[0]}/`;
+      case "dev":
+        return `https://js.dev.paygateway.com/secure_payment/v${version[0]}/`;
+      case "pqa":
+        return `https://js.pqa.paygateway.com/secure_payment/v${version[0]}/`;
+      case "qa":
+        return `https://js.qa.paygateway.com/secure_payment/v${version[0]}/`;
+      case "test":
+        return `https://js.test.paygateway.com/secure_payment/v${version[0]}/`;
+      case "prod":
+        return `https://js.paygateway.com/secure_payment/v${version[0]}/`;
+      case "GP":
+        return result;
+      default:
+        return result;
+    }
+  }
```

Comment:
This should be extracted to an optional gateway action where the Heartland URLs are the default.